app/models/concerns/node_properties.rb

```diff
@@ -72,4 +73,127 @@ def raw_properties=(value)
   rescue JSON::ParserError => exception
     @raw_properties = value
   end
+
+private
+
+  # Private: Adding a :services key to the node properties is a special case.
+  # We only allow a set of known columns in the :services table.
+  # The allowed columms are :name, :port, :product, :protocol, :reason, :state,
+  # :version
+  # Extra columns can be added as an array of hashes with fields :source, :id
+  # and :output inside the :extra column.
+  # These hashes in the :extra column are saved in the :services_extra table.
+  # We check if a row exists in the :services and :services_extra tables
+  # by matching the port and protocol columns.
+  #
+  # value - The new entry (Hash) we want to add to :services
+  #         (and may be :services_extra) tables
+  #
+  # Returns the updated services table
+  def add_service(value)
+    # extract extra info from value
+    value_extra = value.slice(:port, :protocol, :extra)
+    value.slice!(:name, :port, :product, :protocol, :reason, :state, :version)
+
+    merge_service(value)
+
+    unless value_extra[:extra].nil?
```

Comment:
we're using double negative here. This is equivalent to `if value_extra[:extra]`, no?